util/webhook/validator.go

```diff
@@ -167,3 +168,22 @@ func ValidateImmutable(path *field.Path, oldVal, newVal any) *field.Error {
 
 	return nil
 }
+
+// EnsureStringSlicesAreEquivalent returns if two string slices have equal lengths,
+// and that they have the exact same items; it does not enforce strict ordering of items.
+func EnsureStringSlicesAreEquivalent(a []string, b []string) bool {
```

Comment:
This new func is in response to a comment from @nojnhuh [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2718#discussion_r992699827) that the existing `ensureStringSlicesAreEqual` func has a bug (and isn't unit tested).